kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -121,4 +125,38 @@ public Path getLogPath() {
     public Path getDataPath() {
         return dataPath;
     }
+
+    /**
+     * Returns the timestamp of the latest commit of this snapshot.
+     * For an uninitialized snapshot, this returns -1.
+     * <p>
+     * When InCommitTimestampTableFeature is enabled, the timestamp
+     * is retrieved from the CommitInfo of the latest commit which
+     * can result in an IO operation.
+     * <p>
+     * For non-ICT tables, this is the same as the file modification time of the latest commit in
+     * the snapshot.
+     *
+     * @param engine the engine to use for IO operations
+     * @return the timestamp of the latest commit
+     */
+    public long getTimestamp(Engine engine) {
+        if (TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED.fromMetadata(metadata)) {
+            if (!inCommitTimestampOpt.isPresent()) {
+                try {
+                    Optional<CommitInfo> commitInfoOpt = CommitInfo.getCommitInfoOpt(
```

Comment:
Ideally, the ICT should be read part of the protocol and metadata query in LogReplay.java. Can you create an issue to track this improvement?

Reply:
@EstherBear please create an issue for this.

Reply:
Created #3321